src/sentry/static/sentry/app/views/performance/vitalDetail/vitalChart.tsx

```diff
@@ -0,0 +1,298 @@
+    };
+
+    const datetimeSelection = {
+      start: start || null,
+      end: end || null,
+      period: statsPeriod,
+    };
+
+    const vitalThreshold = WEB_VITAL_DETAILS[vitalName].failureThreshold;
+
+    const markLines = [
+      {
+        seriesName: 'Threshold',
+        type: 'line',
+        data: [],
+        markLine: MarkLine({
+          silent: true,
+          lineStyle: {
+            color: theme.textColor,
+            type: 'dashed',
+            width: 1,
+          },
+          label: {
+            show: true,
+            position: 'insideEndBottom',
+            formatter: 'Threshold',
+          },
+          data: [
+            {
+              yAxis: vitalThreshold,
+            } as any, // TODO(ts): date on this type is likely incomplete (needs @types/echarts@4.6.2)
+          ],
+        }),
+      },
+    ];
+
+    const chartOptions = {
+      grid: {
+        left: '10px',
+        right: '10px',
+        top: '40px',
+        bottom: '0px',
+      },
+      seriesOptions: {
+        showSymbol: false,
+      },
+      tooltip: {
+        trigger: 'axis',
+        valueFormatter: tooltipFormatter,
+      },
+      yAxis: {
+        min: 0,
+        max: vitalThreshold,
+        axisLabel: {
+          color: theme.chartLabel,
+          // coerces the axis to be time based
+          formatter: (value: number) => axisLabelFormatter(value, 'p75()'),
+        },
+      },
+    };
+
+    return (
+      <ChartBody>
+        <HeaderTitleLegend>
+          {t(chartTitle || '')}
+          <QuestionTooltip
+            size="sm"
+            position="top"
+            title={t(`The durations shown should fall under the vital threshold.`)}
+          />
+        </HeaderTitleLegend>
+        <ChartZoom
+          router={router}
+          period={statsPeriod}
+          projects={project}
+          environments={environment}
+        >
+          {zoomRenderProps => (
+            <EventsRequest
+              api={api}
+              organization={organization}
+              period={statsPeriod}
+              project={[...project]}
+              environment={[...environment]}
+              start={start}
+              end={end}
+              interval={getInterval(datetimeSelection, true)}
+              showLoading={false}
+              query={query}
+              includePrevious={false}
+              yAxis={yAxis}
+            >
+              {({results, errored, loading, reloading}) => {
+                if (errored) {
+                  return (
+                    <ErrorPanel>
+                      <IconWarning color="gray500" size="lg" />
+                    </ErrorPanel>
+                  );
+                }
+                const colors =
+                  (results && theme.charts.getColorPalette(results.length - 2)) || [];
+
+                const {smoothedResults} = transformEventStatsSmoothed(results);
+
+                const smoothedSeries = smoothedResults
```

Comment:
Should we consider adding better abstractions around creating a smoothed series from another series? I feel like I've seen this code before.

Reply:
doing smoothed series from another series, vs. from results? yeah that would be less code.

Reply:
Yeah. Pass in a series and get a smoothed one back.

Reply:
After thinking about it, I'm actually not that sure, most of the reason for the map below is changing the styling, which is likely something you will do with a smoothed series. If you need to map still anyway, not much point in smoothing the series vs. results.